python/rmm/rmm.py

```diff
@@ -267,6 +267,40 @@ def get_ipc_handle(ary, stream=0):
     )
 
 
+try:
+    import cupy
+
+    class RMMCuPyMemory(cupy.cuda.memory.BaseMemory):
+        def __init__(self, size):
+            self.size = size
+            self.device_id = cupy.cuda.device.get_device_id()
```

Comment:
Note that if RMM is in pool mode using cnmem, and it was not initialized including this device ID, it will be undefined behavior (or likely just fail to allocate).

Reply:
Yeah this was the motivation for the questions above about determining device. Though I think Mads has addressed this by inspecting pointer attributes. Let us know if we missed anything 😉